packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,467 @@
+        new Error(
+          `${ERROR_NO_MODELS_FOUND} ${this.artifactInfo.modelDir}.
+        ${chalk.yellow(
+          'Please visit https://loopback.io/doc/en/lb4/Model-generator.html for information on how models are discovered',
+        )}`,
+        ),
+      );
+    }
+
+    return this.prompt([
+      {
+        type: 'checkbox',
+        name: 'modelNameList',
+        message: PROMPT_MESSAGE_MODEL,
+        choices: modelList,
+        when: this.artifactInfo.modelNameList === undefined,
+      },
+    ])
+      .then(props => {
+        Object.assign(this.artifactInfo, props);
+        debug(`props after model list prompt: ${inspect(props)}`);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during model list prompt: ${err}`);
+        return this.exit(err);
+      });
+  }
+
+  async promptModelId() {
+    if (this.shouldExit()) return false;
+    let idType;
+
+    debug(`Model ID property name from command line: ${this.options.id}`);
+    debug(`Selected Models: ${this.artifactInfo.modelNameList}`);
+
+    if (_.isEmpty(this.artifactInfo.modelNameList)) {
+      return this.exit(new Error(`${ERROR_NO_MODEL_SELECTED}`));
+    } else {
+      // iterate thru each selected model, infer or ask for the ID type
+      for (let item of this.artifactInfo.modelNameList) {
+        this.artifactInfo.modelName = item;
+
+        const prompts = [
+          {
+            type: 'input',
+            name: 'propertyName',
+            message: `Please enter the name of the ID property for ${item}:`,
+            default: 'id',
+          },
+        ];
+
+        // user supplied the id from the command line
+        if (this.options.id) {
+          debug(`passing thru this.options.id with value : ${this.options.id}`);
+
+          idType = this.options.id;
+          /**  make sure it is only used once, in case user selected more
+           * than one model.
+           */
+          delete this.options.id;
+        } else {
+          idType = await this._getModelIdType(item);
+          if (idType === null) {
+            const answer = await this.prompt(prompts);
+            idType = answer.propertyName;
+          }
+        }
+        this.artifactInfo.idType = idType;
+        // Generate this repository
+        await this._scaffold();
+      }
+    }
+  }
+
+  async _scaffold() {
+    if (this.shouldExit()) return false;
+
+    if (this.options.name) {
+      this.artifactInfo.className = utils.toClassName(this.options.name);
+      this.artifactInfo.outFile = utils.getRepositoryFileName(
+        this.options.name,
+      );
+
+      // make sure the name supplied from cmd line is only used once
+      delete this.options.name;
+    } else {
+      this.artifactInfo.className = utils.toClassName(
+        this.artifactInfo.modelName,
+      );
+      this.artifactInfo.outFile = utils.getRepositoryFileName(
+        this.artifactInfo.modelName,
+      );
+    }
+
+    if (debug.enabled) {
```

Comment:
nitpick: this if is not needed as there is nothing process heavy about `outFile`.